File: python3/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        matchmap = dict()
        for i, num in enumerate(nums):
            if num in matchmap:
                return matchmap[num], i
            
            matchmap[target - num] = i

        return -1, -1
        
# @lc code=end

File: cpp/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> map;

        for(int i = 0; i < nums.size(); i++) {
            if(map.contains(nums[i])) {
                return { map[nums[i]], i };
            }

            map[target - nums[i]] = i;
        }

        return {};
    }
};
// @lc code=end

